source/js/components/search-form.js

```diff
@@ -0,0 +1,101 @@
+import fetchUtils from "@degjs/fetch-utils";
+
+const searchForm = () => {
+
+
+    const formEl = document.querySelector('.js-search-form');
+    const layoutEl = document.querySelector('.js-general-content')
+
+    const url = formEl.dataset.api;
+
+    function init() {
+        formEl.addEventListener('submit', e => {
+            e.preventDefault();
+            getData();
+        });
+    }
+
+
+    function getData() {
+        const formEls = formEl.querySelectorAll('.js-value');
```

Comment:
[fix] Your getData method runs as part of an event callback, and inside of it, you are querying the DOM. This means you are re-querying every time getData runs. Can you think of a way to reduce DOM querying here?

Reply:
yep no need to query the DOM every time, I took it out of the getData. Thanks.

Reply:
Where you moved it to doesn't solve the issue though. It queries the DOM every time the ajax request is made.

Reply:
See now it should now query the DOM once?